lib/ohai/mixin/ec2_metadata.rb

```diff
@@ -200,6 +201,13 @@ def fetch_userdata
         response.code == "200" ? response.body : nil
       end
 
+      def fetch_dynamic_data
```

Comment:
also this could be memoized since it looks like you intend it to get reused `@fetch_dynamic_data ||= begin ... end`